packages/shared/src/components/fields/Checkbox.tsx

```diff
@@ -57,12 +59,13 @@ export const Checkbox = forwardRef(function Checkbox(
         { checked: actualChecked, disabled },
       )}
       style={{ transition: 'color 0.1s linear' }}
-      htmlFor={id}
+      htmlFor={id.concat(checkId)}
```

Comment:
We should maybe put this in a variable, to avoid any issue down the line.

Reply:
Updated ✅